### PR TITLE
Adapt error code for `endpoint::Error` and implement the contextual error

### PR DESCRIPTION
**Pull request by YuJuncen**
This PR added the `ErrorCodeExt` implement for the error type.

... and a new extension method `context` for types like `Result<T, E : Into<Error>>`, which is used to add contextual message.

### CODE REVIEW

File: components/backup-stream/src/errors.rs [hunk 1]
```diff
@@ -1,18 +1,19 @@
+use error_code::ErrorCodeExt;
```
Comment (3pointer) [inline components/backup-stream/src/errors.rs]: 🤔

File: components/error_code/src/backup_stream.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    "KV:StreamBackup:",
+
+    ETCD => ("ETCD",
+        "Error during requesting the meta store(etcd)",
+        "Please check the connectivity between TiKV and PD."),
+    PROTO => ("Proto",
+        "Error during decode / encoding protocol buffer messages",
+        "Please check the version of TiKV / BR are compatible, or whether data is corrupted."
+    ),
+    NO_SUCK_TASK => ("NoSuchTask",
```
Comment (3pointer) [inline components/error_code/src/backup_stream.rs]: typo

**Review by 3pointer — APPROVED** [review]
rest LGTM

**Review by joccau — APPROVED** [review]
LGTM

**Pull request merged**